src/main/java/com/mongodb/hibernate/jdbc/MongoResultSet.java

```diff
@@ -0,0 +1,295 @@
+        return bsonValue.isNull() ? 0 : (float) bsonValue.asNumber().doubleValue();
+    }
+
+    @Override
+    public double getDouble(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        BsonValue bsonValue = getBsonValue(columnIndex);
+        return bsonValue.isNull() ? 0 : bsonValue.asNumber().doubleValue();
+    }
+
+    @Override
+    public byte @Nullable [] getBytes(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        BsonValue bsonValue = getBsonValue(columnIndex);
+        return bsonValue.isNull() ? null : bsonValue.asBinary().getData();
+    }
+
+    @Override
+    public @Nullable Date getDate(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        throw new FeatureNotSupportedException("TODO-HIBERNATE-42 https://jira.mongodb.org/browse/HIBERNATE-42");
+    }
+
+    @Override
+    public @Nullable Time getTime(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        throw new FeatureNotSupportedException("TODO-HIBERNATE-42 https://jira.mongodb.org/browse/HIBERNATE-42");
+    }
+
+    @Override
+    public @Nullable Time getTime(int columnIndex, Calendar cal) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        throw new FeatureNotSupportedException("TODO-HIBERNATE-42 https://jira.mongodb.org/browse/HIBERNATE-42");
+    }
+
+    @Override
+    public @Nullable Timestamp getTimestamp(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        throw new FeatureNotSupportedException("TODO-HIBERNATE-42 https://jira.mongodb.org/browse/HIBERNATE-42");
+    }
+
+    @Override
+    public @Nullable Timestamp getTimestamp(int columnIndex, Calendar cal) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        throw new FeatureNotSupportedException("TODO-HIBERNATE-42 https://jira.mongodb.org/browse/HIBERNATE-42");
+    }
+
+    @Override
+    public @Nullable BigDecimal getBigDecimal(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        BsonValue bsonValue = getBsonValue(columnIndex);
+        return bsonValue.isNull()
+                ? null
+                : bsonValue.asDecimal128().decimal128Value().bigDecimalValue();
+    }
+
+    @Override
+    public @Nullable Array getArray(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        throw new FeatureNotSupportedException();
+    }
+
+    @Override
+    public <T> @Nullable T getObject(int columnIndex, Class<T> type) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        throw new FeatureNotSupportedException("In scope of Array / Struct implementation");
```

Comment:
Why add these, if they are to be done later? I see some of them TODO-link to a ticket, which seems like a good reason, but these others do not.

Reply:
Previously we used `NotYetImplementedException` so the message aligns. In last PR @stIncMale suggested renaming `NotYetImplementedException` to `FeatureNotSupportedException`, then it does look weird for we do plan to support it. I removed the confusing message then.

Reply:
But here, I understand the message to mean "this is not yet implemented because it is part of the array/struct impl" - so, it is not an unsupported feature, but something we have not implemented. But then, we seem to not need the override?

Reply:
The decision whether to override is whether we want to support it. We defintiively need to support it. If we remove it from the class, we would lose track and might forget to implement it.

Currently, simply think of `FeatureNotSupportedException` in the non-adapter classes as `NotYetImplementedException`. It is confusing for sure.